salt/states/boto_ec2.py

```diff
@@ -1752,5 +1749,169 @@ def private_ips_absent(name, network_interface_name=None, network_interface_id=N
 
         # there were no changes since we did not attempt to remove ips
         ret['changes'] = {}
+    return ret
+
+
+def secgroup_present(
```

Comment:
I'm not quite sure I follow. Except for instance_present/absent, all other present/absent states in this file are managing resources of EC2 instances. Why would the security-group resources have to be any different?

The instance_present module does have a security_groups-attribute. However running that module with a different set of security-groups (or any resource for that matter) doesn't change the resource of the targeted EC2 instance. That is why issue #38580 exists.